chalk-parse/src/parser.lalrpop

```diff
@@ -217,6 +218,82 @@ Field: Field = {
     }
 };
 
+Scalar: Scalar = {
+    <s:ScalarValue> => Scalar { name: None, value: s },
+};
+
+ScalarValue: ScalarValue = {
+    <Boolean> => ScalarValue::Bool(<>),
+    <Decimal> => <>,
+};
+
+Boolean: bool = <"(true|false)"> => <> == "true";
+
+Decimal: ScalarValue = <s:r"-?[0-9][0-9_]*((u|i)(8|16|32|64|128|size))?"> => {
+    let mut negated = false;
+    let mut digits = String::new();
+    let mut end_type: Option<&str> = None;
+    if s.chars().next().unwrap() == '-' {
+        negated = true;
+        digits.push('-');
+    }
+    for (i, c) in s.chars().enumerate().skip(negated as usize) {
+        if c >= '0' && c <= '9' {
+            digits.push(c);
+        } else if c == 'i' || c == 'u' {
+            end_type = Some(&s[i..]);
+            break;
+        }
+    }
+    if let Some(et) = end_type {
+        // TODO: proper error handling here
+        match et {
+            "u8" => ScalarValue::Uint(UintValue::U8(
+                digits.parse::<u8>().expect("failed to parse u8"),
+            )),
+            "u16" => ScalarValue::Uint(UintValue::U16(
+                digits.parse::<u16>().expect("failed to parse u16"),
+            )),
+            "u32" => ScalarValue::Uint(UintValue::U32(
+                digits.parse::<u32>().expect("failed to parse u32"),
+            )),
+            "u64" => ScalarValue::Uint(UintValue::U64(
+                digits.parse::<u64>().expect("failed to parse u64"),
+            )),
+            "u128" => ScalarValue::Uint(UintValue::U128(
+                digits.parse::<u128>().expect("failed to parse u128"),
+            )),
+            "usize" => ScalarValue::Uint(UintValue::Usize(
+                digits.parse::<usize>().expect("failed to parse usize"),
+            )),
+            "i8" => ScalarValue::Int(IntValue::I8(
+                digits.parse::<i8>().expect("failed to parse i8"),
+            )),
+            "i16" => ScalarValue::Int(IntValue::I16(
+                digits.parse::<i16>().expect("failed to parse i16"),
+            )),
+            "i32" => ScalarValue::Int(IntValue::I32(
+                digits.parse::<i32>().expect("failed to parse i32"),
+            )),
+            "i64" => ScalarValue::Int(IntValue::I64(
+                digits.parse::<i64>().expect("failed to parse i64"),
+            )),
+            "i128" => ScalarValue::Int(IntValue::I128(
+                digits.parse::<i128>().expect("failed to parse i128"),
+            )),
+            "isize" => ScalarValue::Int(IntValue::Isize(
+                digits.parse::<isize>().expect("failed to parse isize"),
+            )),
+            otherwise => unreachable!("Unknown numeric type suffix found: `{}`", otherwise),
+        }
+    } else {
+        // TODO: try various parsers/do whatever rustc does here
```

Comment:
rustc uses inference that falls back to `i32`. I think we should either default to `i32` or else give an error and require people to be explicit.